Library/Homebrew/requirements/minimum_macos_requirement.rb

```diff
@@ -3,18 +3,32 @@
 class MinimumMacOSRequirement < Requirement
```

Comment:
Might be worth renaming this now to be consistent, what do you think?

Reply:
The name of the class is a component of the DSL. The name `MinimumMacOSRequirement` is used by two formulae in core:
```
nordugrid-arc.rb:  depends_on MinimumMacOSRequirement => :yosemite
phantomjs.rb:  depends_on MinimumMacOSRequirement => :lion
```

Reply:
We can add it to `compat` and update them to use `depends_on :macos`

Reply:
Got it. Would you like to suggest a name?

Reply:
`MacOSRequirement`?

Reply:
I've renamed `MinimumMacOSRequirement` to `MacOSRequirement` and added `MinimumMacOSRequirement` to `compat`.